php-classes/Slate/Courses/SectionParticipantsRequestHandler.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Slate\Courses;
+
+
+class SectionParticipantsRequestHandler extends \Slate\RecordsRequestHandler
+{
+    public static $recordClass = SectionParticipant::class;
+
+    public static $accountLevelBrowse = 'Staff';
+    public static $accountLevelWrite = 'Staff';
+    public static $accountLevelRead = 'Staff';
+    public static $accountLevelAPI = 'Staff';
+
+
+    protected static function buildBrowseConditions(array $conditions = [], array &$filterObjects = [])
+    {
+        $conditions = parent::buildBrowseConditions($conditions, $filterObjects);
+
+        if ($Section = static::getRequestedSection()) {
+            $conditions['CourseSectionID'] = $Section->ID;
+            $filterObjects['Section'] = $Section;
+        } elseif ($Term = static::getRequestedTerm()) {
+            $courseSectionIds = array_map(function($section) {
+                return $section->ID;
+            }, Section::getAllByWhere([
+                'TermID' => [
+                    'operator' => 'IN',
+                    'values' => $Term->getContainedTermIDs()
+                ]
+            ]));
+
+            $conditions['CourseSectionID'] = [
+                'operator' => 'IN',
+                'values' => $courseSectionIds
+            ];
+        }
+
+        if ($Student = static::getRequestedStudent()) {
+            $conditions['PersonID'] = $Student->ID;
+            $filterObjects['Student'] = $Student;
+        }
+
+        if (!empty($_REQUEST['cohort'])) {
+            $conditions['Cohort'] = $_REQUEST['cohort'];
+        }
+
+        if (!empty($_REQUEST['role'])) {
+            $conditions['Role'] = [
+                'values' => is_string($_REQUEST['role']) ? explode(',', $_REQUEST['role']) : $_REQUEST['role']
+            ];
+        }
+
+        // apply disabled students filter
+        if (!static::getRequestedStudentDisabledFilter()) {
+          $disabledStudentIds = \DB::allValues(
+              'ID',
+              '
+                  SELECT ID
+                    FROM `%s`
+                  WHERE AccountLevel = "disabled"
+              ',
+              [ \Emergence\People\Person::$tableName ]
+          );
+
+          if (!empty($disabledStudentIds)) {
+              $excludeDisabled = [
+                  'operator' => 'NOT IN',
+                  'values' => $disabledStudentIds
+              ];
+              $existingPersonIDCondition = $conditions['PersonID'];
+
+              if (isset($existingStudentIDCondition)) {
+                  $conditions['PersonID'] = [$existingPersonIDCondition, $excludeDisabled];
```

Comment:
Can you confirm this works? I don't recall mapConditions supporting multiple conditions.

In the case of a condition you want to append but aren't intentionally trying to make it replaceable by downstream filters on the same field, it's best to just not key it by a field at all, like you can see here: https://github.com/JarvusInnovations/emergence-skeleton/blob/develop/php-classes/PeopleRequestHandler.class.php#L77

Reply:
It worked when I tested it, but I like the idea of adding the condition the way your example does it better anyway.  There's much less code if I don't have to worry about a StudentID condition already being set.